contracts/FHE.sol

```diff
@@ -2443,239 +2417,256 @@ library FHE {
+    function decrypt(euint32 input1) internal {
         if (!Common.isInitialized(input1)) {
             input1 = asEuint32(0);
         }
 
-        return euint32.wrap(Impl.decrypt(euint32.unwrap(input1)));
+        euint32.wrap(Impl.decrypt(euint32.unwrap(input1)));
     }
     /// @notice Performs the async decrypt operation on a ciphertext
     /// @dev The decrypted output should be asynchronously handled by the IAsyncFHEReceiver implementation
     /// @param input1 the input ciphertext
-    /// @return the input ciphertext
-    function decrypt(euint64 input1) internal returns (euint64) {
+    function decrypt(euint64 input1) internal {
         if (!Common.isInitialized(input1)) {
             input1 = asEuint64(0);
         }
 
-        return euint64.wrap(Impl.decrypt(euint64.unwrap(input1)));
+        euint64.wrap(Impl.decrypt(euint64.unwrap(input1)));
     }
     /// @notice Performs the async decrypt operation on a ciphertext
     /// @dev The decrypted output should be asynchronously handled by the IAsyncFHEReceiver implementation
     /// @param input1 the input ciphertext
-    /// @return the input ciphertext
-    function decrypt(euint128 input1) internal returns (euint128) {
+    function decrypt(euint128 input1) internal {
         if (!Common.isInitialized(input1)) {
             input1 = asEuint128(0);
         }
 
-        return euint128.wrap(Impl.decrypt(euint128.unwrap(input1)));
+        euint128.wrap(Impl.decrypt(euint128.unwrap(input1)));
     }
     /// @notice Performs the async decrypt operation on a ciphertext
     /// @dev The decrypted output should be asynchronously handled by the IAsyncFHEReceiver implementation
     /// @param input1 the input ciphertext
-    /// @return the input ciphertext
-    function decrypt(euint256 input1) internal returns (euint256) {
+    function decrypt(euint256 input1) internal {
         if (!Common.isInitialized(input1)) {
             input1 = asEuint256(0);
         }
 
-        return euint256.wrap(Impl.decrypt(euint256.unwrap(input1)));
+        euint256.wrap(Impl.decrypt(euint256.unwrap(input1)));
     }
     /// @notice Performs the async decrypt operation on a ciphertext
     /// @dev The decrypted output should be asynchronously handled by the IAsyncFHEReceiver implementation
     /// @param input1 the input ciphertext
-    /// @return the input ciphertext
-    function decrypt(eaddress input1) internal returns (eaddress) {
+    function decrypt(eaddress input1) internal {
         if (!Common.isInitialized(input1)) {
             input1 = asEaddress(address(0));
         }
 
-        return eaddress.wrap(Impl.decrypt(eaddress.unwrap(input1)));
+        Impl.decrypt(eaddress.unwrap(input1));
+    }
+
+    /// @notice Gets the decrypted value from a previously decrypted ebool ciphertext
+    /// @dev This function assumes the ciphertext has already been decrypted. Use getDecryptResultSafe for a safer alternative
```

Comment:
/// @dev This function will revert if the ciphertext is not yet decrypted. Use getDecryptResultSafe for a non-reverting version.

I think this should be updated, it doesn't explain the difference between the safe and unsafe version